docs/polymorphism.rst

```diff
@@ -49,3 +48,30 @@ A class can also be registered manually:
     Discriminator values are written to DynamoDB.
     Changing the value after items have been saved to the database can result in deserialization failures.
     In order to read items with an old discriminator value, the old value must be manually registered.
+
+
+Model Discriminators
+^^^^^^^^^^^^^^^^^^^^
+
+Model classes also support polymorphism through the use of discriminators.
+(Note: currently discriminator attributes cannot be used as the hash or range key of a table.)
+
+.. code-block:: python
+
+    class ParentModel(Model):
+        Meta:
+            table_name = 'polymorphic_table'
+        id = UnicodeAttribute(hash_key=True)
+        cls = DiscriminatorAttribute()
+
+    class FooModel(ParentModel, discriminator='Foo'):
+        foo = UnicodeAttribute()
+
+    class BarModel(ParentModel, discriminator='Bar'):
+        bar = UnicodeAttribute()
+
+.. note::
+
+    Read operations that are performed on a class that has a discriminator value are slightly modified to ensure that only instances of the class are returned.
+    Query and scan operations transparently add a filter condition to ensure that only items with a matching discriminator value are returned.
+    Get and batch get operations will raise a ``ValueError`` if the returned item(s) are not a subclass of the model being read.
```

Comment:
can you add some actual serialized dynamo data here - i was confused about how the discriminator attr was being used until i read your tests below
specifically - the discriminator attribute has the value of `foo` or `bar` which tells us which one to deserialize

this makes a ton of sense based on .... everything
but i feel like the example you've got here could be more clear - maybe change the class names or just add sample serialized documents (like the tests) to make it dumb obvious

Reply:
if the only thing i read was this doc (not the code) i would do it wrong

Reply:
awesome thank you